src/main/java/io/confluent/idesidecar/websocket/resources/WebsocketEndpoint.java

```diff
@@ -50,20 +66,28 @@ public String toString() {
   }
 
   /** Wrapper for a websocket session, with additional state tracking. */
-  public static class WorkspaceWebsocketSession {
+  static class WorkspaceWebsocketSession {
     private final SessionKey key;
     private final Session session;
-    // will grow an Instant createdAt when we start to expire the inactive sessions
-    // after some time -- todo next branch.
+    private final Instant createdAt;
 
     /** Will be null if session has not sent HELLO_WORKSPACE message carrying its process id. */
     @Nullable
     private final WorkspacePid workspacePid;
 
-    public WorkspaceWebsocketSession(Session session, WorkspacePid workspacePid) {
+    WorkspaceWebsocketSession(Session session, WorkspacePid workspacePid, Instant createdAt) {
       key = new SessionKey(session.getId());
       this.session = session;
       this.workspacePid = workspacePid;
+      this.createdAt = createdAt;
+    }
+
```

Comment:
Since WorkspaceWebsocketSession instances are immutable, have a builder-esque method to promote an initial workspace-pid-free session to one with the pid (now smelling "active"), preserving the original's underlying websocket session and createdAt instant.